### PR TITLE
Update api error handling

### PR DESCRIPTION
**Pull request by jason-chan-lf**
- ApiException and ProblemDetails has been updated
- Token Client error handling has been updated to
  - throw ApiException when unable to deserialize response
  - throw ApiException when an undocumented status code is returned

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 2.2.0
```
Comment (steven-xu-lf) [inline CHANGELOG.md:1]: should the pom.xml file version also match this version?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
+
+### Fixes
+- Additional properties have been added to `ProblemDetails` to more accurately represent the API response.
+- **[BREAKING]** Remove `ProblemDetails` inherit from `HashMap<String, Object>`. Undocumented properties can be retrieved using `ProblemDetails.getExtensions()` instead.
+- **[BREAKING]** Remove `ApiException` `response` property. Should use `ApiException` `message` property instead. Additionally, the `ApiException` constructor has been changed to not require the `response` property.
+
```
Comment (steven-xu-lf) [inline CHANGELOG.md:8]: im not sure if this is necessary or not, but should you also indicate that tokenClients have also been updated?

File: CHANGELOG.md (hunk 1 above)
Comment (jason-chan-lf) [reply CHANGELOG.md:1]: I updated the pom to match the version

File: CHANGELOG.md (hunk 2 above)
Comment (jason-chan-lf) [reply CHANGELOG.md:8]: ok i mentioned improved token client error handling

**Review by steven-xu-lf — APPROVED** [review]

**Pull request merged**